arc: cpu_idle: remove sleep workround for nsim_hs_smp

IRISZZW opened the pull request:
In old version nSIM, when cpu is sleeping, no response to inter-processor
interrupt although it's pending and interrupts are enabled(SNPS JIRA
issue P10019563-41294).
Now this has been fixed in nSIM version (2020.09), so we can safely remove it.

nashif commented:
> Now this has been fixed, so we can safely remove it.

which version of nSim does have this fix? Does this work with 

`nSIM, Version: 2019.09 (Build: 013_FREE)`

abrodkin commented:
@nashif it was fixed in 2020.09. I'll check if Free version of that already exists and available.

abrodkin reviewed (changes requested):
Could you please mention in the commit message nSIM version (2020.09) were that fix was integrated?

IRISZZW committed:
arc: cpu_idle: remove sleep workround for nsim_hs_smp

In old version nSIM, when cpu is sleeping, no response to
inter-processor interrupt although it's pending and interrupts
are enabled(SNPS JIRA issue P10019563-41294). Now this has
been fixed in nSIM version (2020.09), so we can safely remove it.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>

abrodkin reviewed (approved):
@nashif I guess we may accept this one regardless newer nSIM version availability since SMP is not [yet] supported by a stand-alone `nsimdrv`, so that will only affect users of full MetaWare toolset and obviously our internal CI.
